completions/spicetify.lua

```diff
@@ -0,0 +1,160 @@
+--------------------------------------------------------------------------------
+-- Clink argmatcher for Spicetify
+--------------------------------------------------------------------------------
+
+function split(inputstr, sep)
+    if sep == nil then
+        sep = "%s"
+    end
+    local t={}
+    for str in string.gmatch(inputstr, "([^"..sep.."]+)") do
+        table.insert(t, str)
+    end
+    return t
+end
+
+function get_extension_names()
+    local handle = io.popen("spicetify.exe path -e")
+    local result = handle:read("*a")
+    handle:close()
+    local paths = split(result, "\n")
+    local names = {}
+    for _, path in ipairs(paths) do
+        local name = path:match("([^\\]+)$")
+        table.insert(names, name)
+    end
+    return names
+end
+
+function get_app_names()
+    local handle = io.popen("spicetify.exe path -a")
+    local result = handle:read("*a")
+    handle:close()
+    local paths = split(result, "\n")
+    local names = {}
+    for _, path in ipairs(paths) do
+        local name = path:match("([^\\]+)$")
+        table.insert(names, name)
+    end
+    return names
+end
+
+local function create_arg(name)
+    return clink.argmatcher():addarg(name):nofiles()
```

Comment:
Regarding this:

> The only thing that doesn't work perfectly is chaining spicetify config arguments for eg:
> `spicetify config inject_css 0 song_page 1 extensions myextensions.js`

Would you like for that work?

You could make it work by changing `create_arg` like this:

```lua
local one_arg = clink.argmatcher():addarg()

local function create_arg(name)
    return clink.argmatcher():addarg(name .. one_arg)
end
```

Then the `:loop(1)` in `config_parser` will have the desired effect.  The reason it doesn't is because of the `:nofiles()` above, which terminates the parser returned by `create_arg` so that it's dead-ended and all further parsing is blocked.

Reply:
That worked perfectly.

Thanks for the explanation, makes sense.